_posts/2023-02-01-user-defined-operators.md

```diff
@@ -0,0 +1,237 @@
+1. `BinaryOperation` and `UnaryOperation` nodes may now have a `function` attribute.
+    If present, the attribute indicates that the operation is actually an invocation of a user-defined
+    operator and specifies the ID of the function used to define it.
+
+This effectively introduces a new way to perform a function call, not represented by a `FunctionCall` node.
+This will affect any tool that needs to detect and follow function calls, for example, to build a
+control flow graph.
+
+### The Future of User-Defined Operators and UDVTs
+Now we know how the new feature works.
+But is it going to change in the future?
+There are actually several planned enhancements you can look forward to.
+
+#### Literal Suffixes
+The very next step is the ability to provide custom literals of user-defined value types.
+The ease of using literal values in expressions is a big part of what makes a type feel native.
+Solidity already provides fractional literals and our intention is to make them usable with custom
+fixed-point number implementations.
+To achieve that we are going to allow defining suffix functions that return UDVT by converting
+literals that are already available in the language:
+
+```solidity
+type UFixed16x2 is uint16;
+
+function uf(int16 mantissa, uint8 exponent) pure suffix returns (UFixed16x2) { ... }
+```
+```solidity
+UFixed16x2 circumference = 2 uf * 3.14 uf * radius;
+```
+
+While the application of a suffix will involve a function call, eventually, with the help of
+compile-time evaluation, literal suffixes are going to become as efficient as native literals.
+
+#### More Operators
+Several operators have been excluded from the initial implementation for the sake of simplicity:
+1. `&&`, `||` and `!` are not necessary for most UDVTs because user-defined comparison operators
+    must return `bool`, for which the built-in boolean operators are sufficient.
+    In case of `&&` and `||`  an additional complication is their short-circuiting behavior.
+1. `<<`, `>>` and `**` have non-uniform arguments.
+    For built-in types, the right-hand side is always an unsigned type.
+1. `=` is one of the few operators that already worked with UDVTs.
+    The new feature is not meant to allow redefining existing operators so providing a custom
+    definition for it will not be possible.
+1. Other assignment operators modify their argument and return a reference.
+    There are currently no references to value types in the language which makes implementing them
+    directly impossible.
+    It may also be desirable to have the compiler provide them automatically whenever the
+    corresponding bitwise or arithmetic operator is defined on the type.
+1. `++` and `--` modify their argument and also have a separate prefix and postfix variants with
+    different semantics.
+
+We are considering enabling them in one of the future releases, provided that the design issues can
+be resolved satisfactorily.
```

Comment:
```suggestion
We are considering enabling them, (except `&&`, `||` and `!`, as mentioned before )
in one of the future releases, provided that the design issues can be resolved satisfactorily.
```

Reply:
I mean, that includes also also `=`. I wanted the list above to indicate which those are instead of repeating them. I'll see if I can reword it.